app/javascript/controllers/attachment_upload_controller.js

```diff
@@ -0,0 +1,78 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+    static targets = ["submitButton", "attachmentsInput"];
+    static values = {
+        buttonText: String
+    }
+
+    connect() {
+        this.element.addEventListener("direct-upload:initialize", event => this.uploadInitialize(event))
+        this.element.addEventListener("direct-upload:start", event => this.uploadStart(event))
+        this.element.addEventListener("direct-upload:progress", event => this.uploadProgress(event))
+        this.element.addEventListener("direct-upload:error", event => this.uploadError(event))
+        this.element.addEventListener("direct-upload:end", event => this.uploadEnd(event))
+
+    }
+
+    uploadInitialize(event) {
+        const { target, detail } = event
+        const { id, file } = detail
+
+        target.insertAdjacentHTML("beforebegin", `
+            <div id="direct-upload-${id}"
+            class="direct-upload direct-upload--pending flex-1 relative mb-1">
+                <div class="flex justify-between mb-1 direct-upload__filename">
+                </div>
+                <div class="w-full bg-gray-200 rounded-full h-2.5 dark:bg-gray-700">
+                    <div class="bg-primary-600 h-2.5 rounded-full" style="width: 0%" id="direct-upload-progress-${id}"></div>
+                </div>
+            </div>
+            `)
+
+        const textColor = localStorage.getItem('theme') == 'dark' ? 'text-blue-700' : 'text-slate-900'
+        target.previousElementSibling.querySelector(`.direct-upload__filename`).innerHTML =
+            `<span class="text-base font-medium ${textColor} dark:text-white">${file.name}</span>
```

Comment:
**nitpick:** So tailwind actually handles all the dark mode stuff for you.  This would probably looked good: `text-gray-500 dark:text-gray-400`.  Can you also replace `text-base font-medium` with just `text-sm`  cleans it up a littl.

Reply:
Done in ef393daa27f8d30f1d8c73ee28fbf04263f9f63f